internal/cutil/aliases.go

```diff
@@ -15,6 +29,9 @@ type SizeT C.size_t
 // unsafe.Pointer but have specific types to help "self document" what the
 // underlying pointer is really meant to represent.
 
+// CPtr is an unsafe.Pointer to C allocated memory
+type CPtr unsafe.Pointer
```

Comment:
As noted in PR #426 I'm on the fence about the value a generic type over unsafe.Pointer gives us. If we know the type that the pointer points to, that'd be a bit different.

Reply:
Well, all I can say is, that it in fact helped _me_ while I was coding the CSlices, that's why I came up with it. I would argue that we should not only rely on `GODEBUG=cgocheck=2` to find go pointers in C memory, and that we are prone to doing that is proven already. ;-) If we use just `unsafe.Pointer` for the Slices for example, nothing stops you from putting an arbitrary pointer in there. With `CPtr` you have to consciously cast, which hopefully makes you think about it if it is in fact a C allocated pointer.

Reply:
OK, if you can get buy in from @nixpanic I'm OK with that rationale. Plus as it's an internal package so we're not locked in to these apis if we need to make a change.

Reply:
@nixpanic can we merge this?